pulsar-client/src/main/java/org/apache/pulsar/client/api/TypedMessageBuilder.java

```diff
@@ -89,6 +89,14 @@
      */
     TypedMessageBuilder<T> key(String key);
 
+    /**
+     * Sets the bytes of the key of the message for routing policy.
+     * Internally the bytes will be base64 encoded.
+     *
+     * @param key routing key for message, in byte array form
+     */
+    TypedMessageBuilder<T> keyBytes(byte[] key);
```

Comment:
any particular reason on naming this `keyBytes` instead of `key`?

Reply:
symmetry with the getKeyBytes() call on Message.